### PR TITLE
:book: Add `release-note` section to `pull_request_template.md`

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
Picking this up from the [Kubernetes template](https://github.com/kubernetes/kubernetes/blob/68f808e6db388b249dae8c6c9dfb078bc0947b49/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L40). This should help when creating release notes to better document breaking changes and action items when upgrading.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
No open issue

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Nice idea, I have a concern though -currently the release notes are autogenerated from the release notes from PR titles which saves a lot of time, and maintainers try to manually highlight breaking changes in the release notes if they're major, or in the book for the most part. 

 This could be misleading for people opening PRs as a result, unless the plan is to change how CAPI release notes are created as well as introducing this field in the template.

**Comment by fabriziopandini** [comment]
I agree with @killianmuldoon.
Might be what we really need instead is a release note highlight, so "normal" PR gets reported in the release notes with the title only as of today, but we have also a way to prepare in advance highlights to add to the release notes when they are required.
Also, despite how do we decide to proceed, the change to the template should also include the change to the release note generation tool so the new information appears in the release note draft

**Comment by CecileRobertMichon** [comment]
FYI in CAPZ we auto generate release notes from the release notes block in PRs using the k8s release note generation tool and it's been working great for us https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/Makefile#L620

**Comment by ykakarap** [comment]
cc @kubernetes-sigs/cluster-api-release-team

**Comment by cpanato** [comment]
i like the release-notes block i think that is useful if we enable the automation to enforce that

**Review by ykakarap — COMMENTED** [review]
+1 to having a mechanism to capture release highlights.

The release notes that we auto-generate can cover the other existing sections of the release notes (⚠️ Breaking Changes, ✨ New Features, 🐛 Bug Fixes, etc). We will definitely benefit from having a mechanism that can be used to generate a draft for highlights which I believe is manually written by maintainers right now.
Although one thing to consider is that the generally features or changes that go into a release highlight are big enough that they span across multiple PRs. So we will have to think about the right UX for developers to be contribute a point to the release highlight section when the associated work could span multiple PRs.

**Comment by joekr** [comment]
Example output:

```
go run hack/tools/release/notes.go 
Changes since testing
---
## :chart_with_upwards_trend: Fun stats
- 4 new commits merged
- 0 breaking changes :warning:
- 1 feature additions ✨
- 0 bugs fixed 🐛

Release notes:
-  The docs were updated to call out breaking change #12345 (#1111)

## :sparkles: New Features
- Introduce ClusterName field to ClusterResourceSetBinding (#7680)

## :seedling: Others
- ClusterClass: don't propagate MD upgrade annotations (#8261)
- Use Lazy Restmapper (#8091)

## :question: Sort these by hand
- docs: Testing some doc updates (#1111)


_Thanks to all our contributors!_ 😊
```

File: hack/tools/release/notes.go [hunk 1]
```diff
@@ -152,8 +153,18 @@ func run() int {
 			commits[last].merge = line
 			continue
 		case line == "":
+		case line == "```release-note":
```
Comment (sbueringer) [inline hack/tools/release/notes.go:156]: I think this tool is currently only parsing the commit message. But it seems like we add the release-note to the PR description not to the commit message.

I would have expected us to call the GitHub API to retrieve the PR description of every commit to be able to handle the release note in the PR descrption

(Maybe I'm missing something)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -5,3 +5,15 @@
 
 **Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
 Fixes #
+
+#### Special notes for your reviewer:
+
+#### Does this PR introduce a user-facing change?
+<!--
+If no, just write "NONE" in the release-note block below.
```
Comment (fabriziopandini) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: Can we have release NONE by default and ask the contributor to change it only if necessary (default to no-op)

File: hack/tools/release/notes.go (hunk 1 above)
Comment (fabriziopandini) [reply hack/tools/release/notes.go:156]: When we change from reading from the PR description, let's make sure to always add the PR number like we do in the other lists e.g. - some test (#8091)

**Comment by vincepri** [comment]
Closing this due to inactivity, feel free to reopen when ready to discuss again.

/close

**Pull request closed without merging**